tilequeue/query.py

```diff
@@ -157,16 +157,17 @@ def __init__(self, conn_info, layer_data, io_pool, n_conn):
 
         self.dbnames = self.conn_info.pop('dbnames')
         self.dbnames_query_index = 0
-        self.sql_conn_pool = DBAffinityConnectionsNoLimit(
-            self.dbnames, n_conn, self.conn_info)
+        self.sql_conn_pool = DatabaseCycleConnectionPool(
+            n_conn, 100, self.dbnames, self.conn_info)
```

Comment:
I wouldn't expect us to ever need to go over the minimum. What we can use for the maximum is the conservative number where we calculate how many io threads we'll need: https://github.com/tilezen/tilequeue/blob/738038db62b9cd33b35b4d9377c5c41d00625e99/tilequeue/command.py#L528

We can just pass that on through to the datafetcher constructor.